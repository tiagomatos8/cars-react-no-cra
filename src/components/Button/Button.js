import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button - The classic button, in different colors, sizes, and states
 * https://bulma.io/documentation/elements/button/
 *
 * @param {string} label text of the button
 * @param {boolean} isRounded makes button rounded
 * @param {boolean} isOutlined makes button outlined
 * @param {string} icon icon inside the button, using fontAwesome
 * @param {string} color color of the button https://bulma.io/documentation/elements/button/#colors
 * @param {string} size size of the button https://bulma.io/documentation/elements/button/#sizes
 * @param {boolean} isFullwidth make button full width
 */
const Button = props => {
  const classNamesObj = {
    isOutlined: props.isOutlined ? 'is-outlined' : '',
    isRounded: props.isRounded ? 'is-rounded' : '',
    color: props.color ? props.color : '',
    label: props.label ? props.label : 'Button',
    size: props.size ? props.size : '',
    isFullwidth: props.isFullwidth ? 'is-fullwidth' : 'Button'
  };

  return (
    <button
      className={`
        Button-component 
        button 
        ${classNamesObj.isOutlined} 
        ${classNamesObj.isRounded} 
        ${classNamesObj.color} 
        ${classNamesObj.size} 
        ${classNamesObj.isFullwidth}
      `}
    >
      {props.icon ? (
        <>
          <span className='icon is-small'>
            <i className={props.icon}></i>
          </span>
          <span>{classNamesObj.label}</span>
        </>
      ) : (
        <>{classNamesObj.label}</>
      )}
    </button>
  );
};

Button.propTypes = {
  label: PropTypes.string,
  isRounded: PropTypes.bool,
  isOutlined: PropTypes.bool,
  icon: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
  isFullwidth: PropTypes.bool
};

export default Button;
