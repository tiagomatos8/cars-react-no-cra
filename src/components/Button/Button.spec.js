import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button Component Test Suite', () => {

  it('Render Button component with no props', () => {
    const component = renderer.create(<Button />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
