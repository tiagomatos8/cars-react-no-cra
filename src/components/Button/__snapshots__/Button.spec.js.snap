// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Test Suite Render Button component with no props 1`] = `
<button
  className="
        Button-component 
        button 
         
         
         
         
        Button
      "
>
  Button
</button>
`;
